hw/dv/data/dsim/dsim.hjson

```diff
@@ -14,22 +14,21 @@
                // UVM
                "+incdir+{UVM_HOME}/src",
                "{UVM_HOME}/src/uvm_pkg.sv",
-               "+acc+rwb -sv_lib {DSIM_HOME}/lib/libuvm_dpi.so",
-
                "-timescale 1ns/1ps",
                "-f {sv_flist}",
                "+incdir+{build_dir}",
                // Suppress following DSim errors and warnings:
                //   EnumMustBePositive - UVM 1.2 violates this
-               "-suppress EnumMustBePositive",
+               "-suppress EnumMustBePositive -warn NeedExplicitStatic",
```

Comment:
Would you please provide a brief comment describing the new tool option?

Reply:
Without this option, the following warning is emitted as an error and causes compilation to fail.
```
=W:[NeedExplicitStatic]:
    Initializers on variables with static lifetime execute once    
    at simulation startup and never thereafter.  Often, users      
    erroneously think such a declaration is automatic.             
    To confirm intent, an explicit 'static' or 'automatic'         
    keyword is required for the following:                         

    ../src/lowrisc_dv_common_ifs_0/clk_rst_if.sv:221:9    done 
```
My opinion is that this message should be a warning by default, and not a hard error. DSim should not fail to compile code that is legal SystemVerilog. I am interested to hear your opinion on this since I plan to discuss it with my colleagues at Metrics. 

The alternate solution is to add the keyword `static` to the declaration of the variable `done`. This situation may recur in other instances as new code is added to the project.

If you are okay with demoting the error to a warning, I'll add a comment explaining the reason for doing so. I thought I should provide the full explanation here first.

Reply:
Indeed according to IEEE 1800-2017 SV LRM:

"""
Variables declared in a static task, function, or procedural block default to a static lifetime and a local scope.
However, **an explicit static keyword shall be required when an initialization value is specified as part of a
static variable’s declaration** to indicate the user’s intent of executing that initialization only once at the
beginning of simulation. 
"""

DSim is probably following the IEEE LRM more closely than other simulators are, and an error in this context makes sense. I would however raise this (with the Metrics devs) anyway to get more opinions on this. 

Can you try changing `bit done = 1'b1;` to `bit done;` in `hw/dv/sv/common_ifs/clk_rst_if.sv`? This should get it to initialize to 0 at run time as opposed to compile time (as a part of the declaration). As such, in this context, it should not make any difference to the simulation. An explicit initialization is not needed (and hence, the error should go away). 

Warnings tend to silently hide glaring issues in our coding, and often result in costly debug overhead. So I would avoid having warnings if possible. Let's remove this demotion and fix the code wherever such an infraction is encountered. 

Thanks, this was educational!

Reply:
Thank you for the discussion! I thought DSim was being overzealous here, I should have looked closer to see that this is actually an LRM infraction. I agree with your analysis, and have changed the code to remove the initialization. Please see the latest commit (https://github.com/lowRISC/opentitan/pull/2485/commits/989ffb4342dbd52105d6b424bf6139859fde6869)

Reply:
Just a quick note from the original author of the offending line: thanks for spotting this. Like Sri, I've just learned something new. Thank you!